From 0faa9d6d8d631f7efe90b25fec4cace02685f5e4 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 25 Jul 2025 08:43:04 +0200
Subject: [PATCH] Revert "backlight: gpio_backlight: add new property
 default-brightness-level"

This reverts commit 2c4e2306ddaa71f26756c35fb9dad2d0f2478ffa.

Change-Id: Ib3e12234cb61a549069895ef79126466a27ec2dc
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/479208
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/leds/backlight/gpio-backlight.yaml          | 9 ---------
 drivers/video/backlight/gpio_backlight.c                 | 7 +------
 2 files changed, 1 insertion(+), 15 deletions(-)

--- a/Documentation/devicetree/bindings/leds/backlight/gpio-backlight.yaml
+++ b/Documentation/devicetree/bindings/leds/backlight/gpio-backlight.yaml
@@ -23,15 +23,6 @@ properties:
     description: enable the backlight at boot.
     type: boolean
 
-  default-brightness-level:
-    description:
-      The default brightness level (index into the array defined by the
-      "brightness-levels" property).
-    $ref: /schemas/types.yaml#/definitions/uint32
-
-dependencies:
-  default-brightness-level: [brightness-levels]
-
 required:
   - compatible
   - gpios
--- a/drivers/video/backlight/gpio_backlight.c
+++ b/drivers/video/backlight/gpio_backlight.c
@@ -53,7 +53,6 @@ static int gpio_backlight_probe(struct p
 	struct backlight_device *bl;
 	struct gpio_backlight *gbl;
 	int ret, init_brightness, def_value;
-	u32 value;
 
 	gbl = devm_kzalloc(dev, sizeof(*gbl), GFP_KERNEL);
 	if (gbl == NULL)
@@ -93,11 +92,7 @@ static int gpio_backlight_probe(struct p
 	else
 		bl->props.power = FB_BLANK_UNBLANK;
 
-	ret = device_property_read_u32(dev, "default-brightness-level", &value);
-	if (!ret && value <= props.max_brightness)
-		bl->props.brightness = value;
-	else
-		bl->props.brightness = 1;
+	bl->props.brightness = 1;
 
 	init_brightness = backlight_get_brightness(bl);
 	ret = gpiod_direction_output(gbl->gpiod, init_brightness);
